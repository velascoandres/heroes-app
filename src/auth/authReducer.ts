export enum AuthActionTypes {
    LOGIN = '[auth] login',
    LOGOUT = '[auth] logout',
}

export interface AuthState {
    username: string;
    logged: boolean;
}

export interface AuthAction {
    type: AuthActionTypes;
}

export interface LoginAction extends AuthAction {
    type: AuthActionTypes.LOGIN;
    payload: {
        name: string;
    };
}

export interface LogoutAction extends AuthAction {
    type: AuthActionTypes.LOGOUT;
}


export const initialAuthState: AuthState = {
    logged: false,
    username: '',
};

export const authReducer = (state: AuthState = initialAuthState, action: AuthAction): AuthState => {
    const { type } = action;
    switch (type) {
        case AuthActionTypes.LOGIN:
            const {payload} = action as LoginAction;
            return {
                ...state,
                username: payload.name,
                logged: true,
            };
        case AuthActionTypes.LOGOUT:
            return {
                ...state,
                username: '',
                logged: false,
            };
        default:
            return {
                ...state,
            };
    }
};
