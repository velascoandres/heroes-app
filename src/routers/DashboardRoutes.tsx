import React, { FC } from 'react';
import { Fragment } from 'react';
import { Navbar } from '../components/ui/Navbar';
import { Switch, Route, Redirect } from 'react-router-dom';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { DcSreen } from '../components/dc/DcSreen';

export const DashboardRoutes: FC = () => {
    return (
        <Fragment>
            <Navbar />

            <div className="container mt-2">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen}></Route>
                    <Route exact path="/heroe/:heroeId" component={HeroScreen}></Route>
                    <Route exact path="/dc" component={DcSreen}></Route>
                    
                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </Fragment>
    );
};
