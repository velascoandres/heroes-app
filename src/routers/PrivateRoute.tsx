import React from 'react'
import { Redirect, Route } from 'react-router-dom';

export interface PrivateRouteProps {
    isAuthenticated: boolean;
    component: React.FC<any>;
    path: string;
}


export const PrivateRoute: React.FC<PrivateRouteProps> = ({isAuthenticated, component: Component, path}: PrivateRouteProps) => {
    
    
    
    return (
       <Route path={path}
        
        component={
            (props: Record<string, any>) => (
                (isAuthenticated) 
                    ?
                 (<Component {...props} />) 
                 : (<Redirect to="/login" />) 
                )
        }
       
       />
           
    );
}
