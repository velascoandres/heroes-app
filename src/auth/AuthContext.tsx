import { createContext } from 'react';
import { AuthState, initialAuthState } from './authReducer';

export const AuthContext = createContext<AuthState>(initialAuthState);
