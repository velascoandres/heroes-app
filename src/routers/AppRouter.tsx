import React, { FC } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter: FC = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={LoginScreen}></Route>
                    <Route path="/" component={DashboardRoutes}></Route>
                </Switch>
            </div>
        </Router>
    );
};
