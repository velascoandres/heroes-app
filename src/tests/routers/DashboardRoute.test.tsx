import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { IAuthContext, AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';


describe('Pruebas en <DashboardRoutes />', () => {

    const authContext: IAuthContext = {
        authState: {
            logged: true,
            username: 'Andres',
        },
        dispatch: jest.fn(),
    };

    test('debe mostrarse correctamente', () => {
        const wrapper = mount(
            <AuthContext.Provider value={authContext}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>    
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Andres');
    });
    

});

