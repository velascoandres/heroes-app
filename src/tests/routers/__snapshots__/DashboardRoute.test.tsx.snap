// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <div
        className="collapse navbar-collapse"
      >
        <div
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
        <div
          className="d-flex"
        >
          <span
            className="nav-item nav-link text-info"
          >
            Andres
          </span>
          <button
            className="btn me-2 nav-item nav-link"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h2>
        Marvel Screen
      </h2>
      <hr />
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInLeft"
          key="marvel-spider"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Spider Man"
                  className="card-img-top"
                  src="./assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInRight"
          key="marvel-captain"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Captain America"
                  className="card-img-top"
                  src="./assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInLeft"
          key="marvel-iron"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Iron Man"
                  className="card-img-top"
                  src="./assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInRight"
          key="marvel-thor"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Thor"
                  className="card-img-top"
                  src="./assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInLeft"
          key="marvel-hulk"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Hulk"
                  className="card-img-top"
                  src="./assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInRight"
          key="marvel-wolverine"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Wolverine"
                  className="card-img-top"
                  src="./assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInLeft"
          key="marvel-daredevil"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Daredevil"
                  className="card-img-top"
                  src="./assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInRight"
          key="marvel-hawkeye"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Hawkeye"
                  className="card-img-top"
                  src="./assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInLeft"
          key="marvel-cyclops"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Cyclops"
                  className="card-img-top"
                  src="./assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 col-sm-12 animate__animated animate__zoomInRight"
          key="marvel-silver"
        >
          <div
            className="card mb-3"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-7 col-sm-12"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img-top"
                  src="./assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-md-5 col-sm-12"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title w-100"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <hr />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    className=""
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    M치s...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
